src/dbt_client/dbtCoreIntegration.ts

```diff
@@ -0,0 +1,521 @@
+            "An unexpected error occured while initializing the dbt project: " +
+              exc +
+              ".",
+          ),
+        );
+        this.telemetry.sendTelemetryError("pythonBridgeInitError", exc);
+      }
+    }
+    this.disposables.push(
+      // when the project config changes we need to re-init the dbt project
+      ...setupWatcherHandler(dbtProfileWatcher, () =>
+        this.rebuildManifest(true),
+      ),
+    );
+  }
+
+  getTargetPath(): string | undefined {
+    return this.targetPath;
+  }
+
+  getModelPaths(): string[] | undefined {
+    return this.modelPaths;
+  }
+
+  getMacroPaths(): string[] | undefined {
+    return this.macroPaths;
+  }
+
+  getPackageInstallPath(): string | undefined {
+    return this.packagesInstallPath;
+  }
+
+  getAdapterType(): string {
+    return this.adapterType;
+  }
+
+  async findAdapterType(): Promise<string | undefined> {
+    return this.python.lock<string>(
+      (python) => python`project.config.credentials.type`,
+    );
+  }
+
+  async rebuildManifest(init: boolean): Promise<void> {
+    try {
+      await this.python.lock(
+        (python) => python`to_dict(project.safe_parse_project(${init}))`,
+      );
+      if (init) {
+        this.adapterType = (await this.findAdapterType()) || "unknown";
+      }
+      this.rebuildManifestDiagnostics.clear();
+    } catch (exc) {
+      if (exc instanceof PythonException) {
+        // dbt errors can be about anything, so we just associate the error with the project file
+        //  with a fixed range
+        this.rebuildManifestDiagnostics.set(
+          Uri.joinPath(this.projectRoot, DBTProject.DBT_PROJECT_FILE),
+          [
+            new Diagnostic(
+              new Range(0, 0, 999, 999),
+              "There is a problem in your dbt project. Compilation failed: " +
+                exc.exception.message,
+            ),
+          ],
+        );
+        this.telemetry.sendTelemetryEvent(
+          "pythonBridgeCannotParseProjectUserError",
+          {
+            error: exc.exception.message,
+            adapter: this.adapterType,
+          },
+        );
+        return;
+      }
+      // if we get here, it is not a dbt error but an extension error.
+      this.telemetry.sendTelemetryError(
+        "pythonBridgeCannotParseProjectUnknownError",
+        exc,
+        {
+          adapter: this.adapterType,
+        },
+      );
+      window.showErrorMessage(
+        extendErrorWithSupportLinks(
+          "An error occured while rebuilding the dbt manifest: " + exc + ".",
+        ),
+      );
+    }
+  }
+
+  async findVersion(): Promise<number[]> {
+    return this.python?.lock<number[]>(
+      (python) => python!`to_dict(project.get_dbt_version())`,
+    );
+  }
+
+  async runModel(command: DBTCommand) {
+    command.addArgument("--profiles-dir");
+    command.addArgument(this.dbtProfilesDir);
+    command.addArgument("--project-dir");
+    command.addArgument(this.projectRoot.fsPath);
```

Comment:
is it possible to extract this common args instead of repeating this for each command?